plugins/catalog-backend-module-incremental-ingestion/src/engine/IncrementalIngestionEngine.ts

```diff
@@ -326,4 +331,42 @@ export class IncrementalIngestionEngine implements IterationEngine {
       removed,
     });
   }
+
+  async onEvent(params: EventParams): Promise<void> {
+    const { topic, eventPayload } = params;
+    if (topic !== this.providerEventTopic) {
+      return;
+    }
+
+    const { logger, provider, connection } = this.options;
+    logger.info(
+      `incremental-engine: Received ${this.providerEventTopic} event`,
+    );
+
+    const payload = eventPayload as TInput;
+
+    if (!provider.deltaMapper) {
+      return;
+    }
+
+    const update = provider.deltaMapper(payload);
+
+    if (update.delta) {
+      await connection.applyMutation({
+        type: 'delta',
+        ...update.delta,
+      });
```

Comment:
I want to emphasize that it's important that whatever this provider does send over the connection, it needs to record in its own tables as well. If a webhook leads to an add here, and then during incremental ingestion it's no longer there, it won't be removed again unless we're recorded this fact when we pushed here.

Reply:
Sorry for the delay, things got busy. I'm *NOW* going to work on this. Also, it occurred to me that I was making deletions too complicated. The end goal, no matter which logical path, is to remove the reference to the entity from the `ingestion_mark_entities` table if we've received a delta for `applyMutation` to run, so all that needs to happen is removal.

Reply:
And done!